apis/src/providers/api_requests.rs

```diff
@@ -1,32 +1,35 @@
 use super::challenges::ChallengeStateSignal;
 use super::games::GamesSignal;
-use super::AuthContext;
+use super::{auth_context, websocket};
 use crate::common::{ChallengeAction, ScheduleAction, TournamentAction};
 use crate::common::{ClientRequest, GameAction};
 use crate::providers::websocket::WebsocketContext;
-use crate::responses::create_challenge_handler;
+use crate::responses::{create_challenge_handler, AccountResponse};
 use hive_lib::{GameControl, Turn};
-use leptos::*;
-use shared_types::{
-    ChallengeId, ChatMessageContainer, GameId, GamesQueryOptions, Takeback, TournamentGameResult,
-    TournamentId,
-};
+use leptos::prelude::*;
+use shared_types::{ChallengeId, ChatMessageContainer, GameId, TournamentGameResult, TournamentId};
 
 #[derive(Clone)]
 pub struct ApiRequests {
     websocket: WebsocketContext,
+    user: Signal<Option<AccountResponse>>,
+    pub challenges: ChallengeStateSignal,
 }
 
-impl Default for ApiRequests {
-    fn default() -> Self {
-        Self::new()
-    }
-}
+#[derive(Clone)]
+pub struct ApiRequestsProvider(pub Signal<ApiRequests>);
```

Comment:
I guess I understand why it's the way it is, but it feels very clunky to me and not really nice from a DX standpoint...